modules/services/sleepwatcher/default.nix

```diff
@@ -0,0 +1,155 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sleepwatcher;
+in
+
+{
+  options = {
+    services.sleepwatcher = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable the sleepwatcher daemon.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.darwin.sleepwatcher;
+        description = "This option specifies the sleepwatcher package to use.";
+      };
+
+      on_sleep = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "pmset sleepnow";
```

Comment:
I'm surprised by this example, does that mean that the script needs to call that to immediately sleep?
If yes it should be mentioned in the description, if not I think the example should be a simple echo like other examples